src/VideoNativeComponent.ts

```diff
@@ -0,0 +1,323 @@
+}>
+
+type BufferConfig = Readonly<{
+  minBufferMs?: number;
+  maxBufferMs?: number;
+  bufferForPlaybackMs?: number;
+  bufferForPlaybackAfterRebufferMs?: number;
+  maxHeapAllocationPercent?: number;
+  minBackBufferMemoryReservePercent?: number;
+  minBufferMemoryReservePercent?: number;
+}>
+
+type SelectedVideoTrack = Readonly<{
+  type?: 'auto' | 'disabled' | 'resolution' | 'index';
+  value?: number;
+}>
+
+type SubtitleStyle = Readonly<{
+  fontSize?: number;
+  paddingTop?: number;
+  paddingBottom?: number;
+  paddingLeft?: number;
+  paddingRight?: number;
+}>
+
+export type OnLoadData = Readonly<{
+  currentTime: number;
+  duration: number;
+  naturalSize: Readonly<{
+    width: number;
+    height: number;
+    orientation: string;
+  }>;
+}>
+
+export type OnLoadStartData = Readonly<{
+  isNetwork: boolean;
+  type: string;
+  uri: string;
+}>
+
+export type OnBufferData = Readonly<{ isBuffering: boolean }>;
+
+
+export type OnProgressData = Readonly<{
+  currentTime: number;
+  playableDuration: number;
+  seekableDuration: number;
+}>
+
+export type OnBandwidthUpdateData = Readonly<{
+  bitrate: number;
+}>;
+
+export type OnSeekData = Readonly<{
+  currentTime: number;
+  seekTime: number;
+  finished: boolean;
+}>
+
+export type OnPlaybackStateChangedData = Readonly<{
+  isPlaying: boolean;
+}>
+
+export type OnTimedMetadataData = Readonly<{
+  metadata: ReadonlyArray<Readonly<{
+    value?: string
+    identifier?: string
+  }>>
+}>
+
+
+export type OnAudioTracksData = Readonly<{
+  audioTracks: ReadonlyArray<Readonly<{
+    index?: number
+    title?: string
+    language?: string
+    bitrate?: number
+    type?: string
+    selected?: boolean
+  }>>
+}>
+
+export type OnTextTracksData = Readonly<{
+  textTracks: ReadonlyArray<Readonly<{
+    index?: number
+    title?: string
+    language?: string
+    /**
+     * iOS only supports VTT, Android supports all 3
+     */
+    type?: 'srt' | 'ttml' | 'vtt'
+    selected?: boolean
+  }>>
+}>
+
+export type OnVideoTracksData = Readonly<{
+  videoTracks: ReadonlyArray<Readonly<{
+    trackId?: number
+    codecs?: string
+    width?: number
+    height?: number
+    bitrate?: number
+    selected?: boolean
+  }>>
+}>
+
+export type OnPlaybackData = Readonly<{
+  playbackRate: number;
+}>;
+
+export type OnExternalPlaybackChangeData = Readonly<{
+  isExternalPlaybackActive: boolean;
+}>
+
+export type OnGetLicenseData = Readonly<{
+  licenseUrl: string;
+  contentId: string;
+  spcBase64: string;
+}>
+
+export type OnPictureInPictureStatusChangedData = Readonly<{
+  isActive: boolean;
+}>
+
+export type OnReceiveAdEventData = Readonly<{
+  event: string;
+}>
+
+export type OnVideoErrorData = Readonly<{
+  error: string;
+}>
+
+export type OnAudioFocusChangedData = Readonly<{
+  hasFocus: boolean;
+}>
+
+export type NativeVideoResizeMode = 'ScaleNone' | 'ScaleToFill' | 'ScaleAspectFit' | 'ScaleAspectFill';
+export interface VideoNativeProps extends ViewProps {
+  src: VideoSrc;
```

Comment:
Missing '?'
Client app can put src=undefined to stop the player

Reply:
Fixed ✅